Lib/test/test_ssl.py

```diff
@@ -151,6 +151,26 @@ def data_file(*name):
 OP_CIPHER_SERVER_PREFERENCE = getattr(ssl, "OP_CIPHER_SERVER_PREFERENCE", 0)
 OP_ENABLE_MIDDLEBOX_COMPAT = getattr(ssl, "OP_ENABLE_MIDDLEBOX_COMPAT", 0)
 
+# Ubuntu has patched OpenSSL and changed behavior of security level 2
+# see https://bugs.python.org/issue41561#msg389003
+def is_ubuntu():
+    try:
+        # assume that any references of "ubuntu" implies Ubuntu-like distro
+        with open("/etc/os-release") as f:
```

Comment:
I'm deliberately not using ``platform.freedesktop_os_release()`` here. The function is 3.10-only and I want to backport the fix to 3.7 to 3.9.

Reply:
I noticed that `platform.freedesktop_os_release()` opens with `encoding="utf-8"` and also checks `/usr/lib/os-release`. Are either of those steps necessary here?

Reply:
I'll add ``encoding="utf-8"``. The check for ``/usr/lib/os-release`` is not necessary on Ubuntu. ``/etc/os-release`` always exists.

I don't want to spend more time on this issue than necessary. I have already spent way to much time on a platform that I don't even use. The approach isn't perfect but good enough to fix testing on Ubuntu. 

20.04, 20.10, and 21.04 are definitely affected. The man page https://manpages.ubuntu.com/manpages/hirsute/man3/SSL_CTX_set_security_level.3ssl.html is a good indicator. If you see ``On Ubuntu, TLS versions below 1.2 are not permitted.`` then the custom policy applies.